src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java

```diff
@@ -192,27 +193,17 @@ void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithWelshFlagIsY() throws
     }
 
     @Test
-    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsY_200()
+    void shouldReturnFailureForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsY_200()
         throws JsonProcessingException {
-        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
-            "AAA1?flag-type=party&available-external-flag=Y",
-            CaseFlag.class,
-            path
-        );
-        assertEquals(1, response.getFlags().get(0).getFlagDetails().size());
-        List<FlagDetail> parentFlags = response.getFlags().get(0).getFlagDetails();
-        List<String> sampleDefaultStatus = Arrays.asList("Active", "Requested");
-        for (FlagDetail parentFlag : parentFlags) {
-            if (parentFlag.getParent()) {
-                assertNotNull(parentFlag);
-                parentFlag.getChildFlags().forEach(childFlag -> {
-                    if (!childFlag.getName().equalsIgnoreCase("Other") && !childFlag.getParent()) {
-                        assertEquals(true, childFlag.getExternallyAvailable());
-                        assertTrue(sampleDefaultStatus.contains(childFlag.getDefaultStatus()));
-                    }
-                });
-            }
-        }
+        Exception exception = assertThrows(UnrecognizedPropertyException.class, () -> {
+            commonDataApiClient.retrieveCaseFlagsByServiceId(
+                "AAA1?flag-type=party&available-external-flag=Y",
+                CaseFlag.class,
+                path
+            );
+        });
+        assertNotNull(exception);
+        assertTrue(exception.getLocalizedMessage().contains("Data not found"));
     }
 
```

Comment:
i dont see test cases for prd roles scenario and and also there are no assertions for available-external-flag requests for parent flag and all child flags.

Reply:
added integration tests